JOSESwift/Sources/Algorithms.swift

```diff
@@ -115,3 +115,12 @@ public enum HMACAlgorithm: String {
         }
     }
 }
+
+/// An algorithm for compressing the plain text before encryption.
+/// List of [supported compression algorithms](https://www.iana.org/assignments/jose/jose.xhtml#web-encryption-compression-algorithms)
+///
+/// - Deflate: [DEF](https://tools.ietf.org/html/rfc7516#section-4.1.3)
+public enum CompressionAlgorithm: String {    
```

Comment:
Since `NONE` is really for internal use only (although it would not break anything if the user would use it) I think we're fine with keeping it undocumented. 👍